sdk/src/Core/Amazon.Runtime/Credentials/BasicAWSCredentials.cs

```diff
@@ -53,6 +53,7 @@ public BasicAWSCredentials(string accessKey, string secretKey) : this (accessKey
         /// <param name="accountId">The account id for the credentials. The account id is your 12 digit account number with no hyphens. For example: 123456789012</param>
         public BasicAWSCredentials(string accessKey, string secretKey, string accountId)
         {
+            if (string.IsNullOrEmpty(accountId)) FeatureIdSources.Add(UserAgentFeatureId.RESOLVED_ACCOUNT_ID);
```

Comment:
Is there a better place we can check an account ID was resolved? Maybe in the signer? (where we have access to the immutable credentials object and would not need to update multiple credential providers)

This if is also inverted, right? `accountId` should _not_ be empty for us to track the feature ID.